pkg/configuration/reader/converter.go

```diff
@@ -163,15 +163,25 @@ func ConvertGateways(gateways []*Gateway, trench *nspAPI.Trench) []*nspAPI.Gatew
 		resGateways = append(resGateways, &nspAPI.Gateway{
 			Name:       gateway.Name,
 			Address:    gateway.Address,
+			IpFamily:   gateway.IPFamily,
+			Protocol:   gateway.Protocol,
 			RemoteASN:  gateway.RemoteASN,
 			LocalASN:   gateway.LocalASN,
 			RemotePort: uint32(gateway.RemotePort),
 			LocalPort:  uint32(gateway.LocalPort),
-			IpFamily:   gateway.IPFamily,
-			Bfd:        gateway.BFD,
-			Protocol:   gateway.Protocol,
 			HoldTime:   uint32(gateway.HoldTime),
-			Trench:     trench,
+			Bfd:        gateway.BFD,
+			BfdSpec: func() *nspAPI.Gateway_BfdSpec {
+				if !gateway.BFD {
+					return nil
+				}
+				return &nspAPI.Gateway_BfdSpec{
+					MinTx:      uint32(gateway.MinTx),
```

Comment:
Maybe MinTx, MinRx, Multiplier and HoldTime should be set to uint32 in /pkg/configuration/reader/model.go (this change would require some modifications in the operator)

Some gateway fields could be empty in the configmap, would default values be needed?

Reply:
In case of BFD parameters there are default values on the Operator side for both BGP and Static protocols. So they will never be empty.
That's needed because in BIRD BFD default values are different than the ones we intend to use.

On the other hand, if any of the BFD attributes took the variables default value, it should not cause any problem, as the FE will ignore that, thus BIRD's default setting for that attribute will take effect.